source/sc/2.4.4.html.md.erb

```diff
@@ -4,27 +4,80 @@ title: 2.4.4 Link Purpose (In Context) (A)
+
+## How to check
+
+The accessible name is the name used by assistive technologies such as screen readers and voice control tools. To find the accessible name of each link, you can use:
+
+* your browser's developer tools
+* a screen reader
+
+Check that each link describes the link's purpose. If the link text itself is not enough to understand the purpose, then also check its immediate context, such as the same paragraph, list item, table cell or an associated table header.
+
+Automated tools such as Axe can detect links with no text, but they cannot determine whether link text is appropriate when it exists.
+
+[How to test in detail for 2.4.4 Link Purpose (in context)](https://github.com/alphagov/wcag-primer/wiki/2.4.4)
+
+## Good examples
+
+### A self-explanatory link
+
+<figure>
+    <%= image_tag "../images/244/244-pass-self-explanatory-link.png", { :alt => 'A "Financial support" link after a paragraph of text.' } %>
+</figure>
+
+Ideally, all links should make sense out of context. In this case, the "Financial support" link does not depend on the previous text for you to understand it.
+
+### A link that makes sense in context
+
+<figure>
+    <%= image_tag "../images/244/244-pass-link-in-context.png", { :alt => 'The text "We provide support to help you manage your finances" with a link on the word "support".' } %>
+</figure>
+
+While the link text of "support" is not very specific in itself, the surrounding context makes the link meaningful.
 
 ## Common mistakes
 
-*   A link has text that does not indicate its purpose;
-*   A link points to the same destination as another link, but uses different link text;
-*   A link points to a unique destination, but uses the same text as other links;
-*   A link has text that depends on nearby content to be understood, but that content is not automatically identified by assistive technologies;
-*   A link uses a CSS background image, and has no visible accessible name.
+### Links with no text
+
+A link fails this criterion if it is an image with no associated text. In this example, the Twitter and Search icons are described as "Unlabeled" when using a screen reader.
+
+<figure>
+    <%= image_tag "../images/244/244-fail-links-with-no-name.png", { :alt => 'Instagram and Search icons' } %>
+        <figcaption>The accessible names of these links are unreadable symbols rather than text.</figcaption>
+</figure>
+
+### Links out of context
+
+The following link has generic text of "Read more" and is not in the same paragraph as the wording leading up to it, meaning it has no immediate context.
+
+<figure>
+    <%= image_tag "../images/244/244-fail-read-more.png", { :alt => 'A "read more" link after a paragraph of text introducing an article.' } %>
+    <figcaption>In this case, there is no link in the code between the paragraph and the link.</figcaption>
+</figure>
+
```

Comment:
These are much better and with much clearer images.

I wonder if we should also provide "how to avoid these common mistakes" to make the guide more helpful?
For example, add text labels (or at least `aria-label`) to the icon links.
The read more one is a bit more complex as the best option would possibly be linking up the title of the article and removing the read more entirely.